funcbench/main.go

```diff
@@ -0,0 +1,354 @@
+	"golang.org/x/oauth2"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type gitClient struct {
+	owner  string
+	branch string
+	repo   string
+}
+type gitHubClient struct {
+	owner            string
+	repo             string
+	latestCommitHash string
+	prNumber         int
+	client           *github.Client
+}
+
+type benchmarkTester struct {
+	bechRegex       string
+	home            string
+	raceFlagEnabled bool
+}
+
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "benchmark result posting and formating tool.\n-i location of github hook file (even.json)")
+	app.HelpFlag.Short('h')
+	input := app.Flag("input", "path to event.json").Short('i').Default("/github/workflow/event.json").String()
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	data, err := ioutil.ReadFile(*input)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Temporary fix for the new Github actions time format, as go-github is not capable of parsing it. This makes the time stamps unusable.
+	txt := string(data)
+	reg := regexp.MustCompile("(.*)\"[0-9]+/[0-9]+/2019 [0-9]+:[0-9]+:[0-9]+ [AP]M(.*)")
+	txt = reg.ReplaceAllString(txt, "$1\"2019-06-11T09:26:28Z$2")
+	data = []byte(txt)
+	// End of the temporary fix.
+
+	event, err := github.ParseWebHook("issue_comment", data)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	var (
+		ghClient  *gitHubClient
+		gitClient *gitClient
+		prNumber  int
+	)
+	switch eventType := event.(type) {
+	case *github.IssueCommentEvent:
+		ghClient = newGitHubClient(eventType)
+		gitClient, err = newGitClient(eventType)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		prNumber = *eventType.GetIssue().Number
+
+	default:
+		log.Fatalln("only issue_comment event is supported")
+	}
+	benchTest, err := newBenchmarkTester()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	logLink := fmt.Sprintf("Full logs at: https://github.com/%s/%s/commit/%s/checks", ghClient.owner, ghClient.repo, ghClient.latestCommitHash)
+
+	if err := os.Chdir(os.Getenv("GITHUB_WORKSPACE")); err != nil {
+		log.Fatalln(err)
+	}
+
+	if err := gitClient.cloneRepository(); err != nil {
+		log.Fatalln(err)
+	}
+
+	if err := gitClient.checkoutPR(prNumber); err != nil {
+		comment := fmt.Sprintf("Switch to a pull request branch failed. %s", logLink)
+		if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+			log.Fatalf("Error posting a comment for `checkoutToPullRequest` command execution error. checkoutToPullRequest err:%v, postComment err:%v", err, postCommentErr)
+		}
+		log.Fatalln(err)
+	}
+
+	newBenchmark, err := benchTest.execBenchmark()
+	if err != nil {
+		comment := fmt.Sprintf("Go bench test for this pull request failed. %s", logLink)
+		if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+			log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+		}
+		log.Fatalln(err)
+	}
+
+	if err := gitClient.revertPRChanges(); err != nil {
+		log.Fatalln(err)
+	}
+
+	oldBenchmark, err := benchTest.execBenchmark()
```

Comment:
```suggestion
	branchResults, err := benchTest.execBenchmark()
```

Reply:
or maybe call it before and after

just old and new sounds a bit strange. Not a blocker and can leave it as is if you prefer.